addons/binding/org.openhab.binding.wifiled/src/main/java/org/openhab/binding/wifiled/handler/FadingWiFiLEDDriver.java

```diff
@@ -59,12 +74,29 @@ public void init() throws IOException {
             dtoState = LEDStateDTO.valueOf(s.state, s.program, s.programSpeed, s.red, s.green, s.blue, s.white,
                     s.white2);
             power = (s.state & 0x01) != 0;
-            currentTargetState = InternalLedState.fromRGBW(s.red, s.green, s.blue, s.white, s.white2);
+            currentFaderState = currentState = InternalLedState.fromRGBW(s.red, s.green, s.blue, s.white, s.white2);
         } catch (IOException ignored) {
         }
     }
 
     @Override
+    public void dispose() {
+        waiterFuture.shutdown();
+        faderFuture.shutdown();
+        try {
+            if (!waiterFuture.awaitTermination((fadeDurationInMs / fadeSteps) * 2, TimeUnit.MILLISECONDS)) {
+                waiterFuture.shutdownNow();
+            }
+            if (!faderFuture.awaitTermination((fadeDurationInMs / fadeSteps) * 2, TimeUnit.MILLISECONDS)) {
+                faderFuture.shutdownNow();
+            }
+        } catch (InterruptedException e) {
+            logger.warn("InterruptedException", e);
```

Comment:
Given you are disposing you can ignore this exception, you especially don't have to `log.warn` it